### PR TITLE
docs: Proof-read compactor docs

### PR DESCRIPTION
**Pull request by mhutter**
* [ ] I added CHANGELOG entry for this change.
* [x] Change is not relevant to the end user.

## Changes

* Proof-read the compactor docs and removed some common english mistakes to improve readability

## Verification

I read it :)

### CODE REVIEW

File: docs/components/compact.md [hunk 1]
```diff
@@ -274,24 +275,24 @@ usage: thanos compact [<flags>]
 Continuously compacts blocks in an object store bucket.
 
 Flags:
-      --block-meta-fetch-concurrency=32  
+      --block-meta-fetch-concurrency=32
```
Comment (saswatamcode) [inline docs/components/compact.md:278]: Hi! Thanks for the PR! 

These whitespaces are actually generated by [mdox](https://github.com/bwplotka/mdox), which pastes CLI output directly into these code blocks. Docs CI will fail if these are removed.

You can run `make docs` to correct these and any other markdown formatting issues. :)

Comment (mhutter) [reply docs/components/compact.md]: Thanks for the hint, I yanked the 1st commit from the PR and ran `make docs` to fix some issues with my changes

File: docs/components/compact.md [hunk 2]
```diff
@@ -46,57 +46,57 @@ This is because `external_labels` are added by the Prometheus which produced the
...
 
-This rule, means also that there could be a problem when both compacted and non compacted blocks are being uploaded by sidecar. This is why "upload compacted" flag is still under a separate `--shipper.upload-compacted` flag that helps to ensure that compacted blocks are uploaded before anything else. The singleton rule is also why local Prometheus compaction has to be disabled in order to use sidecar with upload option. Use hidden `--shipper.ignore-unequal-block-size` to override this check (on your own risk).
+This rule also means that there could be a problem when both compacted and non-compacted blocks are being uploaded by a sidecar. This is why the "upload compacted" function still lives under a separate `--shipper.upload-compacted` flag that helps to ensure that compacted blocks are uploaded before anything else. The signleton rule is also why local Prometheus compaction has to be disabled in order to use Thanos Sidecar with the upload option. Use - at your own risk! - the hidden `--shipper.ignore-unequal-block-size` flag to disable this check.
```
Comment (GiedriusS) [inline docs/components/compact.md:57]:
```suggestion
This rule also means that there could be a problem when both compacted and non-compacted blocks are being uploaded by a sidecar. This is why the "upload compacted" function still lives under a separate `--shipper.upload-compacted` flag that helps to ensure that compacted blocks are uploaded before anything else. The singleton rule is also why local Prometheus compaction has to be disabled in order to use Thanos Sidecar with the upload option. Use - at your own risk! - the hidden `--shipper.ignore-unequal-block-size` flag to disable this check.
```

File: docs/components/compact.md [hunk 3]
```diff
@@ -46,57 +46,57 @@ This is because `external_labels` are added by the Prometheus which produced the
...
 
-In both systems, series with the same labels are merged together. In prometheus, merging samples is **naive**. It works by deduplicating samples within exactly the same timestamps. Otherwise samples are added in sorted by time order. Thanos also support a new penalty based samples merger and it is explained in [Deduplication](#vertical-compaction-use-cases).
+In both systems, series with the same labels are merged together. In Prometheus, merging samples is **naive**. It works by deduplicating samples within exactly the same timestamps. Otherwise samples are merged and sorted by timestamp. Thanos also supports a new penalty based samples merging strategy, which is explained in [Deduplication](#vertical-compaction-use-cases).
 
-> **NOTE:** Both Prometheus and Thanos default behaviour is to fail compaction if any overlapping blocks are spotted. (For Thanos, within the same external labels).
+> **NOTE:** Both Prometheus' and Thanos' default behaviour is to fail compaction if any overlapping blocks are spotted. (For Thanos, within the same external labels).
```
Comment (GiedriusS) [inline docs/components/compact.md:73]:
```suggestion
> **NOTE:** Both Prometheus' and Thanos' default behaviour is to fail compaction if any overlapping blocks are spotted. (For Thanos, with the same external labels).
```

File: docs/components/compact.md [hunk 4]
```diff
@@ -46,57 +46,57 @@ This is because `external_labels` are added by the Prometheus which produced the
...
-  * `penalty` deduplication is when same series data is **logically duplicated**. For example, it comes from the same application, but scraped by two different Prometheus-es. Ideally this requires more complex deduplication algorithms. For example one that is used to [deduplicate on the fly on the Querier](query.md#run-time-deduplication-of-ha-groups). This is common case when Prometheus HA replicas are used. You can enable this deduplication via `--deduplication.func=penalty` flag.
+* **Races** between multiple compactions, for example multiple Thanos compactors or between Thanos and Prometheus compactions. While this will cause extra computational overhead for Compactor it's safe to enable vertical compaction for this case.
+* **Backfilling**. If you want to add blocks of data to any stream where there already is existing data for this time range, you will need to enable vertical compaction.
```
Comment (GiedriusS) [inline docs/components/compact.md:80]:
```suggestion
* **Backfilling**. If you want to add blocks of data to any stream where there already is existing data for some time range, you will need to enable vertical compaction.
```

File: docs/components/compact.md [hunk 5]
```diff
@@ -46,57 +46,57 @@ This is because `external_labels` are added by the Prometheus which produced the
...
+* **Offline deduplication** of series. It's very common to have the same data replicated into multiple streams. We can distinguish two common strategies for deduplications, `one-to-one` and `penalty`:
+  * `one-to-one` deduplication is when multiple series (with the same labels) from different blocks for the same time range have **exactly** the same samples: Same values and timestamps. This is very common when using [Receivers](receive.md) with replication greater than 1 as receiver replication copies samples exactly (same timestamps and values) to different receive instances.
+  * `penalty` deduplication is when the same data is **duplicated logically**, i.e. the same application is scraped from two different Prometheis. This usually requires more complex deduplication algorithms. or example one that is used to [deduplicate on the fly on the Querier](query.md#run-time-deduplication-of-ha-groups). This is common case when Prometheus HA replicas are used. You can enable this deduplication strategy via the `--deduplication.func=penalty` flag.
```
Comment (GiedriusS) [inline docs/components/compact.md:83]:
```suggestion
  * `penalty` deduplication is when the same data is **duplicated logically**, i.e. the same application is scraped from two different Prometheis. This usually requires more complex deduplication algorithms. For example, one that is used to [deduplicate on the fly on the Querier](query.md#run-time-deduplication-of-ha-groups). This is a common case when Prometheus HA replicas are used. You can enable this deduplication strategy via the `--deduplication.func=penalty` flag.
```

**Review by GiedriusS — DISMISSED** [review]
Some suggestions but otherwise LGTM :+1: thank you, good docs are important!

**Review by GiedriusS — APPROVED** [review]
Thank you! :muscle:

**Pull request merged**